test/transaction_test.exs

```diff
@@ -25,6 +25,32 @@ defmodule TransactionTest do
     {:ok, [pid: pid]}
   end
 
+  @tag mode: :transaction
+  test "commit log entries", context do
+    assert transaction(fn conn ->
+      P.query(conn, "SELECT 42", [])
+    end, log: &send(self(), &1))
+
+    assert_receive %DBConnection.LogEntry{} = entry
+    assert {:ok, %DBConnection{}, %Postgrex.Result{command: :begin}} = entry.result
```

Comment:
not sure if all these tests are worth keeping, but that's what I used to verify it.